docs/DeveloperGuide.md

```diff
@@ -145,33 +159,33 @@ Each `UiView` has a mini lifecycle. `viewDidLoad` is run after `render` is calle
 - Set UI component values (e.g. using `setText` on an FXML `Text` object).
 - Attach subviews and propagate the chain.
 
-#### 2.1.2. UiStore ####
+#### 3.2.2. UiStore ####
 
 **API** : [`UiStore.java`](../src/main/java/seedu/toluist/ui/UiStore.java)
 
 `UiStore` holds the data to be used by the `UI`. An example would be the task data to be displayed to the user.
 
-#### 2.1.3. Reactive nature of the UI ####
+#### 3.2.3. Reactive nature of the UI ####
 
 To keep the UI predictable and to reduce the number of lines of codes used to dictate how the UI should change based on state changes, we make use of reactive programming in our UI. You can declare how the UI should be rendered based solely on the states held by the `UiStore`.
 
 The diagram below shows how the UI reacts when an add command is called. The UI simply needs to display all the tasks available in the `UiStore`, without knowing what was the exact change.
 
 <img src="images/UiSequence.png" width="600"><br>
-_Figure 2.1.3 : Interactions Inside the UI for the `add study` Command_
+_Figure 3.2.3 : Interactions Inside the UI for the `add study` Command_
 
 The reactive approach is borrowed from modern Javascript front-end frameworks such as [React.js](https://facebook.github.io/react/) and [Vue.js](https://vuejs.org/v2/guide/reactivity.html).
 
-#### 2.2 Dispatcher component
+### 3.3. Dispatcher component
 
 **API** : [`Dispatcher.java`](../src/main/java/seedu/toluist/dispatcher/Dispatcher.java)
 
 `Dispatcher` acts like a router in a traditional Web MVC architecture. On receiving new input from the UI, `Dispatcher` decides which `Controller` is the best candidate to handle the input, then instantiates and asks the `Controller` object to execute the command.
 
-### 2.3. Controller component
+### 3.4. Controller component
 
 <img src="images/ControllerClassDiagram.png" width="600"><br>
-_Figure 2.3.1 : Structure of the Controller Component_
+_Figure 3.3 : Structure of the Controller Component_
 
```

Comment:
Do you want to change the figure naming also? Or leave as 3.3 since there is no figure for the previous section?